### PR TITLE
fix(major): Add session endpoint support.

### PR DESCRIPTION
**Pull request by ser-0xff**
## Description

Add session endpoint support.

## How Has This Been Tested?

Unit tests.

## Minimal checklist:

- [X] I have performed a self-review of my own code 
- [ ] I have added `DocC` code-level documentation for any public interfaces exported by the package
- [ ] I have added unit and/or integration tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: Sources/ConsulServiceDiscovery/Consul.swift [hunk 1]
```diff
@@ -85,6 +85,38 @@ public final class Consul: Sendable {
...
+                let promise = impl.makePromise(of: Void.self)
+                impl.request(method: .PUT, uri: "/v1/agent/check/register", body: requestBody, handler: ResponseHandlerVoid(promise))
+                return promise.futureResult
+            } catch {
+                return impl.makeFailedFuture(error)
+            }
+        }
+
+        /// Deregister check
+        /// - Parameter check: check description
```
Comment (dimlio) [inline Sources/ConsulServiceDiscovery/Consul.swift:108]:
```suggestion
        /// - Parameter checkId: check identifier
```

File: Sources/ConsulServiceDiscovery/Consul.swift [hunk 2]
```diff
@@ -85,6 +85,38 @@ public final class Consul: Sendable {
...
+        /// - Returns: EventLoopFuture<Void> to deliver result
+        /// [apidoc] https://developer.hashicorp.com/consul/api-docs/agent/check#deregister-check
+        ///
+        public func deregisterCheck(_ checkId: String) -> EventLoopFuture<Void> {
+            impl.logger.debug("deregister check \(checkId)")
+            let promise = impl.makePromise(of: Void.self)
+            let uri = "/v1/agent/check/register/\(checkId)"
```
Comment (dimlio) [inline Sources/ConsulServiceDiscovery/Consul.swift:115]:
```suggestion
            let uri = "/v1/agent/check/deregister/\(checkId)"
```
?

File: Sources/ConsulServiceDiscovery/Consul.swift [hunk 3]
```diff
@@ -409,14 +460,137 @@ public final class Consul: Sendable {
...
+        ///
+        public func create(_ session: Session, inDatacenter datacenter: String? = nil) -> EventLoopFuture<String> {
+            struct ResponseHandler: ConsulResponseHandler {
+                private let promise: EventLoopPromise<String>
+
+                init(_ promise: EventLoopPromise<String>) {
+                    self.promise = promise
+                }
+
+                func processResponse(_ buffer: ByteBuffer, withIndex: Int?) {
```
Comment (dimlio) [inline Sources/ConsulServiceDiscovery/Consul.swift:503]:
```suggestion
                func processResponse(_ buffer: ByteBuffer, withIndex _: Int?) {
```
unused?

File: Sources/ConsulServiceDiscovery/Consul.swift (hunk 1 above)
Comment (ser-0xff) [reply Sources/ConsulServiceDiscovery/Consul.swift]: Fixed.

File: Sources/ConsulServiceDiscovery/Consul.swift (hunk 2 above)
Comment (ser-0xff) [reply Sources/ConsulServiceDiscovery/Consul.swift]: Fixed.

File: Sources/ConsulServiceDiscovery/Consul.swift (hunk 3 above)
Comment (ser-0xff) [reply Sources/ConsulServiceDiscovery/Consul.swift]: Fixed.

**Review by dimlio — APPROVED** [review]

**Pull request merged**